testsuite/features/step_definitions/setup_steps.rb

```diff
@@ -258,19 +258,54 @@
 end
 
 Then(/^I should see "([^"]*)" at least (\d+) minutes after I scheduled an action$/) do |text, minutes|
-  # TODO: is there a better way then page.all ?
   elements = all('div', text: text)
-  raise ScriptError, "Text #{text} not found in the page" if elements.nil?
 
-  match = elements[0].text.match(/#{text}\s*(\d+\/\d+\/\d+ \d+:\d+:\d+ (AM|PM)+ [^\s]+)/)
-  raise ScriptError, "No element found matching text '#{text}'" if match.nil?
+  # Pattern Breakdown (Strict MM/DD/YY, flexible timezone name):
+  # (\d{2}\/\d{2}\/\d{2})  -> MM/DD/YY (STRICT)
+  # \s+ -> One or more spaces
+  # (\d{1,2}:\d{2}:\d{2}) -> HH:MM:SS
+  # \s* -> Zero or more spaces (for potential non-breaking space)
+  # (AM|PM) -> AM or PM
+  # \s+ -> One or more spaces
+  # (\w+) -> Capture the Timezone (e.g., UTC, GMT, CEST, etc.)
+  date_time_pattern = /(\d{2}\/\d{2}\/\d{2})\s+(\d{1,2}:\d{2}:\d{2})\s*(AM|PM)\s+(\w+)/
+  match = elements[0].text.match(/#{text}\s*#{date_time_pattern}/)
+  raise ScriptError, "The date/time pattern (MM/DD/YY HH:MM:SS AM/PM TZ) was not found after '#{text}'" if match.nil?
+
+  # match[1] is the Date (MM/DD/YY)
+  # match[2] is the Time (HH:MM:SS)
+  # match[3] is the AM/PM
+  # match[4] is the Timezone name (TZ)
+  time_string = "#{match[1]} #{match[2]} #{match[3]} #{match[4]}"
+  format_string = '%m/%d/%y %I:%M:%S %p %Z'
 
-  text_time = DateTime.strptime(match.captures[0].to_s, '%m/%d/%C %H:%M:%S %p %Z')
-  raise ScriptError, 'Time the action was scheduled not found in memory' unless defined?($moments) && $moments['schedule_action']
+  begin
+    displayed_time = DateTime.strptime(time_string, format_string)
+  rescue ArgumentError => e
+    raise ScriptError, "Failed to parse time string '#{time_string}' with format '#{format_string}': #{e.message}"
+  end
 
-  initial = $moments['schedule_action']
-  after = initial + (Rational(1, 1440) * minutes)
-  raise ScriptError, "#{text_time} is not #{minutes} minutes later than '#{initial}'" unless (text_time + Rational(1, 1440)) >= after
+  scheduled_time = $moments['schedule_action']
+  unless scheduled_time.is_a?(Time) || scheduled_time.is_a?(DateTime)
+    raise ScriptError, 'Time the action was scheduled not found or not a time object in $moments["schedule_action"]'
+  end
+
+  initial_dt =
```

Comment:
Just a cosmetic change. We can put it in just one line
```suggestion
  initial_dt = scheduled_time.is_a?(Time) ? scheduled_time.to_datetime : scheduled_time
```